src/darwin/Framework/CHIP/CHIPOperationalCredentialsDelegate.mm

```diff
@@ -203,11 +203,12 @@ static BOOL isRunningTests(void)
         return CHIP_ERROR_INTERNAL;
     }
 
+    uint16_t rcacBufLen = static_cast<uint16_t>(std::min(rcac.size(), static_cast<size_t>(UINT16_MAX)));
+    PERSISTENT_KEY_OP(fabricId, kOperationalCredentialsRootCertificateStorage, key,
+        haveRootCert = mStorage->SyncGetKeyValue(key, rcac.data(), rcacBufLen) == CHIP_NO_ERROR);
```

Comment:
Probably better to set `haveRootCert` ( stack variable) to false, then if `!mForceRootCertRegeneration` try doing the `SyncGetKeyValue` and set to the result of that, so we don't bother with the `SyncGetKeyValue` if we're going to ignore its result anyway?

So something like:
```
bool haveRootCert = false;
if (!mForceRootCertRegeneration) {
    // Get from storage, set haveRootCert
}

if (!haveRootCert) {
  // Generate it
}

mForceRootCertGeneration = false;
````

Reply:
Ah ok. Makes sense.